src/rules/no_misused_new.rs

```diff
@@ -28,6 +28,35 @@ impl LintRule for NoMisusedNew {
   fn code(&self) -> &'static str {
     "no-misused-new"
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Disallows defining constructors for interfaces or new for classes
+
+Specifying a constructor for an interface or defining a `new` method for a class
+is incorrect and should be avoided.
+    
+### Invalid:
+```typescript
+class C {
+  new(): C;
+}
+
+interface I {
+  constructor(): void;
+}
+```
+
+### Valid:
+```typescript
+class C {
+  constructor() {}
+}
+interface I {
+  new (): C;
+}
```

Comment:
```suggestion
class C {
  constructor() {}
}

interface I {
  new (): C;
}
```


a very minor nit, but it'd be nice to have a blank line between `C` and `I` as the `Invalid` example has 👍